cmd/wallet-sdk-gomobile/api/api.go

```diff
@@ -25,56 +27,79 @@ type KeyHandleWriter interface {
 	Rotate(keyType, keyID string, opts string) (*KeyHandle, error)
 	// Import will import privKey into the KMS storage for the given keyType then returns the new key id and
 	// the newly persisted KeyHandle.
-	// privKey possible types are: *ecdsa.PrivateKey and ed25519.PrivateKey. // TODO: Determine how these restrictions work.
+	// privKey possible types are: *ecdsa.PrivateKey and ed25519.PrivateKey.
+	// TODO: Determine how these restrictions work.
 	// kt possible types are signing key types only (ECDSA keys or Ed25519)
 	// opts allows setting the keysetID of the imported key using WithKeyID() option. If the ID is already used,
 	// then an error is returned. // TODO: Format of opts to be determined.
 	// Returns: a key handle, which contains both the new key ID and actual private key bytes.
 	// An error/exception will be returned/thrown if there is an import failure (key empty, invalid, doesn't match
 	// keyType, unsupported keyType or storing of key failed)
 	// TODO: Consider renaming this method to avoid keyword collision and automatic renaming to _import in Java
-	Import(privKey *KeyHandle, keyType, opts string) (*KeyHandle, error)
+	Import(privateKey *KeyHandle, keyType, opts string) (*KeyHandle, error)
 }
 
+// A KeyHandleReader represents a type that is capable of performing certain operations related to reading key data.
 type KeyHandleReader interface {
-	// GetKeyHandle key handle for the given keyID
-	// Returns the private key bytes
-	//  - error if failure
-	GetKeyHandle(keyID string) ([]byte, error)
+	// GetKeyHandle return the key handle for the given keyID
+	// Returns:
+	//  - The private key handle
+	//  - Error if failure
+	GetKeyHandle(keyID string) (*KeyHandle, error)
 	// Export will fetch a key referenced by id then gets its public key in raw bytes and returns it.
 	// The key must be an asymmetric key.
 	// Returns:
 	//  - A key handle, which contains both the key type and public key bytes
-	//  - error if it fails to export the public key bytes
+	//  - Error if it fails to export the public key bytes
 	Export(keyID string) (*KeyHandle, error)
 }
 
+// DIDCreator defines the method required for a type to create DID documents.
 type DIDCreator interface {
 	// Create creates a new DID Document.
 	// It returns a DID Document Resolution.
 	Create(didDocument []byte) ([]byte, error)
```

Comment:
didDocument as request??

Reply:
https://github.com/trustbloc/agent-sdk/blob/main/cmd/wallet-js-sdk/src/did/didmanager.js#L71

Reply:
Updated the PR. I change the method signature to take in a `CreateDIDOpts` in a similar style as some of the other various methods in the SDK introduced by this PR. I only added a `Method` option for now since I'm not sure exactly what will be required for the various DID methods - when I add a `did:Key` DIDCreator implementation I can update this as needed. (Or if you have a good idea of what exactly we'll need, let me know and I'll add it now)